flink-table/flink-table-planner/src/test/java/org/apache/flink/table/planner/plan/rules/logical/ProjectWatermarkAssignerTransposeRuleTest.java

```diff
@@ -159,4 +159,12 @@ void transposeWithDuplicateColumns() {
     void transposeWithWatermarkWithMultipleInput() {
         util.verifyRelPlan("SELECT a FROM UdfTable");
     }
+
+    @Test
+    void transposeWithHints() {
```

Comment:
Nit: Is there a particular reason for adding the test in this RuleTest instead of in others like StateTtlHintTest?

Reply:
no strong reason, just because the issue is related to watermark assignment
thanks for the feedback

moved to StateTtlHintTest